presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardCompactionManager.java

```diff
@@ -175,8 +176,8 @@ private void discoverShards()
                 continue;
             }
 
-            Long temporalColumnId = metadataDao.getTemporalColumnId(tableId);
-            if (temporalColumnId != null) {
+            OptionalLong temporalColumnId = metadataDao.getTemporalColumnId(tableId).getTemporalColumnId();
```

Comment:
The SQL object method will return null if there are no rows (the `ResultSetMapper` is never called because there are no rows to map).